Panic with outer join and fill + usePrevious

Seems there is something wrong in how either fill consumes data in multiple buffers or in how outer join produces multiple buffers that causes a panic. See the test below, it panics as written, however if you uncomment  `debug.slurp` the test will pass. 


```
package universe_test


import "testing"
import "internal/debug"
import "array"
import "join"

testcase fill_time_previous {
    left =
        array.from(rows: [{_time: 2022-01-01T00:00:00Z, _value: 2}, {_time: 2022-01-01T00:00:15Z, _value: 1}])
            |> map(fn: (r) => ({r with myStart: r._time}))

    right = array.from(rows: [{_time: 2022-01-01T00:00:30Z, _value: 0}])
    got =
        join.full(
            left: left,
            right: right,
            on: (l, r) => l._time == r._time,
            as: (l, r) => {
                time = if exists l._time then l._time else r._time
                value = if exists l._value then l._value else r._value

                return {_time: time, value: value, myStart: l.myStart}
            },
        )
            // Adding debug.slurp avoids the panic and the test passes
            //|> debug.slurp()
            |> fill(column: "myStart", usePrevious: true)
    want =
        array.from(
            rows: [
                {_time: 2022-01-01T00:00:00Z, myStart: 2022-01-01T00:00:00Z, value: 2},
                {_time: 2022-01-01T00:00:15Z, myStart: 2022-01-01T00:00:15Z, value: 1},
                {_time: 2022-01-01T00:00:30Z, myStart: 2022-01-01T00:00:15Z, value: 0},
            ],
        )

    testing.diff(got, want)
}
```

Panic stack

```
panic: interface conversion: interface {} is int64, not values.Time

goroutine 50 [running]:
github.com/influxdata/flux/stdlib/universe.(*fillTransformation).fillTimeColumn(0xc001b1a540, 0xc001b21440?, 0xc0025f0f20, {0x2214b20?, 0xc003eb1a40?})
	/projects/flux/stdlib/universe/fill.gen.go:192 +0x2af
github.com/influxdata/flux/stdlib/universe.(*fillTransformation).fillColumn(0x2225f50?, 0xc001b5cc80?, {0x2222198?, 0xc001b21d80?}, 0x1?, {0x2214b20?, 0xc003eb1a40?})
	/projects/flux/stdlib/universe/fill.gen.go:31 +0x15a
github.com/influxdata/flux/stdlib/universe.(*fillTransformation).fillTable(0xc001b1a540, 0xc001b5ce40, {0x2225f50, 0xc001b5cc80}, 0x1, 0xc0026475e0?)
	/projects/flux/stdlib/universe/fill.go:433 +0x465
github.com/influxdata/flux/stdlib/universe.(*fillTransformation).Process.func1.1({0x2225f50?, 0xc001b5cc80?})
	/projects/flux/stdlib/universe/fill.go:257 +0x3f
github.com/influxdata/flux/execute/table.(*BufferedTable).Do(0xc002e3ad90?, 0xc0026476e0)
	/projects/flux/execute/table/buffered.go:69 +0x19b
github.com/influxdata/flux/stdlib/universe.(*fillTransformation).Process.func1({0x0?, 0xc003576600?}, 0xc001b5ce40)
	/projects/flux/stdlib/universe/fill.go:256 +0xd8
github.com/influxdata/flux/internal/execute/table.StreamWithContext.func1()
	/projects/flux/internal/execute/table/stream.go:109 +0xa6
created by github.com/influxdata/flux/internal/execute/table.StreamWithContext
	/projects/flux/internal/execute/table/stream.go:106 +0x21d
exit status 2
```

--- a/libflux/go/libflux/buildinfo.gen.go
+++ b/libflux/go/libflux/buildinfo.gen.go
@@ -501,7 +501,7 @@ var sourceHashes = map[string]string{
 	"stdlib/universe/integral_test.flux":                                                          "7206d881f059f0e6009fe3f0cfeb18b5fcec9ba192b607f63d375541851fcb85",
 	"stdlib/universe/join_across_measurements_test.flux":                                          "4362913c852740d1e55dc13d2b9ac798ffde4263a4b78ca0c96bd5a1333278ad",
 	"stdlib/universe/join_agg_test.flux":                                                          "94c4de77895501644c9fd100dda93d85a76e35b39e97bd801f59f3e6179b337c",
-	"stdlib/universe/join_fill_panic_test.flux":                                                   "4b1706d3ac001bb9bffe5f26bfa7d26e3149a4a15b048cb48cd7657eb4643f2b",
+	"stdlib/universe/join_fill_panic_test.flux":                                                   "60976aa9a75b79ccb03d2403df3f644f8f70abf0da71d066a3d03bcc1cc9904f",
 	"stdlib/universe/join_mismatched_schema_test.flux":                                            "4a50ae64612961f8fc9decc0ad87305eb937931d9608f8a9796cedad35699ff8",
 	"stdlib/universe/join_missing_on_col_test.flux":                                               "477ddf1b38200ff1d3ff63b020e5a89c6b0c9adb4f468d69f56176669c4363e3",
 	"stdlib/universe/join_panic_test.flux":                                                        "1397a1c71a1efb7efe0a5a3e057e7543c59cf405707e637c0073f48b6c28cc03",

--- a/libflux/go/libflux/buildinfo.gen.go
+++ b/libflux/go/libflux/buildinfo.gen.go
@@ -501,6 +501,7 @@ var sourceHashes = map[string]string{
 	"stdlib/universe/integral_test.flux":                                                          "7206d881f059f0e6009fe3f0cfeb18b5fcec9ba192b607f63d375541851fcb85",
 	"stdlib/universe/join_across_measurements_test.flux":                                          "4362913c852740d1e55dc13d2b9ac798ffde4263a4b78ca0c96bd5a1333278ad",
 	"stdlib/universe/join_agg_test.flux":                                                          "94c4de77895501644c9fd100dda93d85a76e35b39e97bd801f59f3e6179b337c",
+	"stdlib/universe/join_fill_panic_test.flux":                                                   "4b1706d3ac001bb9bffe5f26bfa7d26e3149a4a15b048cb48cd7657eb4643f2b",
 	"stdlib/universe/join_mismatched_schema_test.flux":                                            "4a50ae64612961f8fc9decc0ad87305eb937931d9608f8a9796cedad35699ff8",
 	"stdlib/universe/join_missing_on_col_test.flux":                                               "477ddf1b38200ff1d3ff63b020e5a89c6b0c9adb4f468d69f56176669c4363e3",
 	"stdlib/universe/join_panic_test.flux":                                                        "1397a1c71a1efb7efe0a5a3e057e7543c59cf405707e637c0073f48b6c28cc03",

--- a/stdlib/universe/fill.gen.go
+++ b/stdlib/universe/fill.gen.go
@@ -210,7 +210,7 @@ func (t *fillTransformation) fillTimeColumn(arr *array.Int, fillValue *interface
 		}
 	}
 	if t.spec.UsePrevious && !fillValueNull {
-		*fillValue = fillValueTime
+		*fillValue = values.Time(fillValueTime)
 	}
 	return b.NewArray()
 }

--- a/stdlib/universe/fill.gen.go.tmpl
+++ b/stdlib/universe/fill.gen.go.tmpl
@@ -45,7 +45,7 @@ func (t *fillTransformation) fill{{.Name}}Column(arr *{{.ArrowType}}, fillValue 
         }
     }
     if t.spec.UsePrevious && !fillValueNull {
-        *fillValue = fillValue{{.Name}}
+        *fillValue = {{if eq .Name "Time"}}values.Time(fillValue{{.Name}}){{else}}fillValue{{.Name}}{{end}}
     }
     return b.NewArray()
 }

--- a/stdlib/universe/join_fill_panic_test.flux
+++ b/stdlib/universe/join_fill_panic_test.flux
@@ -1,0 +1,40 @@
+package universe_test
+
+
+import "testing"
+import "internal/debug"
+import "array"
+import "join"
+
+testcase fill_time_previous {
+    left =
+        array.from(rows: [{_time: 2022-01-01T00:00:00Z, _value: 2}, {_time: 2022-01-01T00:00:15Z, _value: 1}])
+            |> map(fn: (r) => ({r with myStart: r._time}))
+
+    right = array.from(rows: [{_time: 2022-01-01T00:00:30Z, _value: 0}])
+    got =
+        join.full(
+            left: left,
+            right: right,
+            on: (l, r) => l._time == r._time,
+            as: (l, r) => {
+                time = if exists l._time then l._time else r._time
+                value = if exists l._value then l._value else r._value
+
+                return {_time: time, value: value, myStart: l.myStart}
+            },
+        )
+            // Adding debug.slurp avoids the panic and the test passes
+            // |> debug.slurp()
+            |> fill(column: "myStart", usePrevious: true)
+    want =
+        array.from(
+            rows: [
+                {_time: 2022-01-01T00:00:00Z, myStart: 2022-01-01T00:00:00Z, value: 2},
+                {_time: 2022-01-01T00:00:15Z, myStart: 2022-01-01T00:00:15Z, value: 1},
+                {_time: 2022-01-01T00:00:30Z, myStart: 2022-01-01T00:00:15Z, value: 0},
+            ],
+        )
+
+    testing.diff(got, want)
+}

--- a/stdlib/universe/join_fill_panic_test.flux
+++ b/stdlib/universe/join_fill_panic_test.flux
@@ -8,7 +8,12 @@ import "join"
 
 testcase fill_time_previous {
     left =
-        array.from(rows: [{_time: 2022-01-01T00:00:00Z, _value: 2}, {_time: 2022-01-01T00:00:15Z, _value: 1}])
+        array.from(
+            rows: [
+                {_time: 2022-01-01T00:00:00Z, _value: 2},
+                {_time: 2022-01-01T00:00:15Z, _value: 1},
+            ],
+        )
             |> map(fn: (r) => ({r with myStart: r._time}))
 
     right = array.from(rows: [{_time: 2022-01-01T00:00:30Z, _value: 0}])

